packages/remix-server-runtime/server.ts

```diff
@@ -55,7 +55,7 @@ export const createRequestHandler: CreateRequestHandlerFunction = (
     let { unstable_dev } = build.future;
     if (
       mode === "development" &&
-      unstable_dev !== false &&
+      typeof unstable_dev === "object" &&
```

Comment:
Since `object` is vague (arrays, `Date`s, and a bunch of stuff has `typeof` set to `object`), let's check for `boolean` instead:

```
typeof unstable_dev !== 'boolean' &&
```